stories/translator/doc-elements.tsx

```diff
@@ -0,0 +1,121 @@
+import { Text } from "@lifesg/react-design-system/text";
+import { EyeIcon, PencilIcon, TabletIcon } from "@lifesg/react-icons";
+import {
+    FrontendEngine,
+    IFrontendEngineData,
+} from "@lifesg/web-frontend-engine";
+import { useEffect, useRef, useState } from "react";
+import { FormBuilder, IFormBuilderMethods } from "src/form-builder";
+import {
+    ContentWrapper,
+    IconButton,
+    IconWrapper,
+    ViewWrapper,
+    SchemaPreview,
+} from "./doc-elements.styles";
+
+interface IProps {
+    data?: IFrontendEngineData;
+}
+
+const FormPreview = ({ data }: IProps) => {
+    return (
+        <ViewWrapper>
+            <Text.H2>Generate Form</Text.H2>
+            {data && <FrontendEngine data={data} />}
+        </ViewWrapper>
+    );
+};
+
+const SchemaView = ({ data }: IProps) => {
+    return (
+        <ViewWrapper>
+            <Text.H2>Generate Schema</Text.H2>
+            <SchemaPreview>
+                <pre>{JSON.stringify(data, null, 2)}</pre>
+            </SchemaPreview>
+        </ViewWrapper>
+    );
+};
+
+export const DocElement = () => {
+    // =========================================================================
+    // CONST, STATE, REFS
+    // =========================================================================
+    const formBuilderRef = useRef<IFormBuilderMethods>(null);
+    const [pageMode, setPageMode] = useState<string>("form-builder-mode");
+    const [schema, setSchema] = useState<IFrontendEngineData | null>(null);
+
+    // =========================================================================
+    // EVENT HANDLERS
+    // =========================================================================
+    const handleFormBuilderButton = (event) => {
+        event.preventDefault();
+        setPageMode("form-builder-mode");
+    };
+
+    const handleFormPreviewButton = (event) => {
+        event.preventDefault();
+        setPageMode("preview-mode");
+    };
+
+    const handleSchemaViewButton = (event) => {
+        event.preventDefault();
+        setPageMode("schema-mode");
+    };
```

Comment:
- `event.preventDefault();` is not needed now that we added `type="button"` for each of the button
- declare typing for the event, we avoid leaving them untyped because they will default to `any`